nebula/ui/components/VPNTutorialPopups.qml

```diff
@@ -0,0 +1,289 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.5
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import Mozilla.VPN 1.0
+import compat 0.1
+
+
+Item {
+
+    /*
+        TODOs
+
+        - Handle banner alerts
+        - Handle conflicts with update required
+
+        - Should tutorials timeout?
+        - Better UI in landscape mode
+        - Respond to targetElement y position changes
+        - Animation polish on tooltip open
+
+        - Update event filter
+            - Restrict wheel events
+            - Consider restricting key commands to allowed items
+
+        - Second tutorial nits:
+            - Toggle checkbox status with space bar does not trigger next step
+            - System preferences is not immediately focused, it's possible to tab to the back arrow and get in a weird state
+    */
+
+
+    // targetElement is set to `parent` here to get around `Cannot call method ... of undefined` warnings
+    // and is reset before the tutorial is opened in onTooltipNeeded()
+    property var targetElement: parent
+    onTargetElementChanged: pushFocusToTargetElement()
+
+    id: root
+    objectName: "tutorialUiRoot"
+
+    anchors.fill: parent
+    visible: tutorialTooltip.visible || tutorialPopup.opened
+
+    Popup {
+        property alias tooltipText: tooltipText.text
+        property int notchHeight: VPNTheme.theme.windowMargin
+        property bool tooltipPositionedAboveTargetElement: false
+
+        id: tutorialTooltip
+
+        closePolicy: Popup.NoAutoClose
+        onClipChanged: VPNTutorial.stop();
+        verticalPadding: VPNTheme.theme.windowMargin
+        horizontalPadding: VPNTheme.theme.windowMargin
+        focus: true
+
+        y: {
```

Comment:
Should we use `Qt.binding(function() { ...` ?

Reply:
Oooh good question. This may fix the `Respond to targetElement y position changes` TODO.